### DISCUSSION TITLE
Bring back custom frequencies

### DISCUSSION THREAD

**Discussion by azxcqer** [iSoron/uhabits#961] [Feature Requests] [answered]
I have a lot of habits and plenty of them are using custom frequencies (6 times every 8 days and similar). The new update the option to have custom frequencies is gone but as far as I can see they are still supported because none of my habits broke, I just can't change the frequencies to these things. 

It is not really a deal breaker, since I could easily extrapolate the frequencies (looking at my example for instance using the 22 times every month wouldn't make that much of a difference in the long run) but I also think it wouldn't be too hard and wouldn't complicate things too much.

I mean, I really loved the last release and it really added and improved soo much but I'm missing this feature

**Comment by McCio** [comment]
Strongly agree to make them come back.

I'll add that custom frequencies is the only way to encode the "maximum days without performing the habit", while converting to longer timespan would make it possible to _self-fake_ reaching the habit when it wasn't actually followed.

For a clearer example, converting a "2 every 4 days" to "15 every month" would consider valid the whole month when performing the action of the habit 15 days consecutively, while the other 15 you didn't perform it at all.
Instead, keeping the "2/4 days" you allow only 2 consecutive days of not following the habit before the app starts considering _failed_ days: in the example, there would be 13 failed days.

**Comment by hiqua** [comment]
I'm also missing the old custom frequencies: it used to be possible to specify once in a year (365 days). It seems that it's no longer possible, as the input only allows two decimals, so 99 days maximum.

This kind of habit stretches the definition, but I think a lot of people start using this app track things other than regular habits when they're happy about the basic features for normal habits. I think we shouldn't put artificial constraints on these use-cases.

An example would be periodic medical appointments.

**Comment by Kerobyte** [comment]
I had multiple habits that were over a year long and I think they should be reimplemented. For example appliance maintenance is a habit that can span years, replacing washing machine or vacuum cleaner belts etc. At the very least the two digit limit on "Every x days" should be lifted.

**Comment by LordJon** [comment]
Currently using Google Play version 1.8.12 stable release. I've tried the new version on a secondary device with an backup. It looks very promising except missing these custom frequencies is a reason to stay on the older version for me.
I hope it can be reimplemented.

**Reply by LordJon** [reply]
A small question related to this. The Google Play version currently is at 2.0.3. I have disabled auto updates so I can stay on the older version. Will this gives problems when 2.1 gets released and I decide to update from 1.8.12 to 2.1?
I can imagine this being a problem since the database isn't backwards compatible due to major changes.

**Reply by hiqua** [reply]
That would be a bug, but I encourage you to do a full backup before upgrading if that's a concern (and fill an issue if you do encounter such a bug!).

**Reply by LordJon** [reply]
I'm making regular backups so that won't be a problem I think, thanks. Sounds like it should work, we'll see when 2.1 is released.

Thanks for a the hard work so far. This app and the philosophy behind it has kinda changed my life. I've never been able to grow new habits this good as I'm doing now, especially the mind set is different as the philosophy (and score) behind the app encourages to keep trying. Looking forward to the future with all new features being implemented. Thanks all!

**Comment by iSoron** [comment, marked as answer]
Implemented in `dev` and should become available in Loop 2.1.

**Reply by azxcqer** [reply]
awesome! thanks :)

any estimative on when 2.1 will be released?

**Reply by hiqua** [reply]
My guess is: when https://github.com/iSoron/uhabits/milestone/13 is empty.

**Reply by luisvalenzuelar** [reply]
@hiqua site is empty :) hope new version comes soon!!

**Reply by jimfarrand** [reply]
@iSoron Is there any way we can help getting this release out? Love this app, and it's dead impressive that it's so stable it can go so long without releases, but it would be great to get this functionality out into the wild.

**Comment by end-ichi** [comment]
For those who don't want to wait for the new release, you can currently set arbitrary frequencies with this method:

1. Do a full export.
2. Open the database file, e.g. by copying it to PC and using SQLite Browser
3. Go to "Browse Data" tab and select "Habits" table
4. To get "x out of y days", set freq_num=x and freq_den=y (double click, enter value, click apply)
5. Click "Write Changes"
6. Move the file back to your phone and import into the app.

**Reply by azxcqer** [reply]
thanks! that's really useful!

**Reply by noccer** [reply]
I also found this really useful, thanks for sharing. 

I installed the SQLite extension for visual studio code, connected to the database, wrote a SQL script, saved it then did the `SQLite: Run Query` command in VSCode. Then executed on extension the DB with no issues. Then I reimported the updated .db file back into the app and I now have my custom interval working just fine. A handy workaround though I agree with previous posters that it would really be nice to have UI access to the Numerator and Denominator re-enabled so you can do something say like 1 in 180 days etc (i.e. half year habit)

**Reply by end-ichi** [reply]
@noccer it is already re-enabled in version 2.1.

**Reply by noccer** [reply]
Thanks but how do I get it, I'm only getting V2.0.3 in the play store. Cheers ☘️

**Reply by noccer** [reply]
☝️ @wobbba 

forgot to tag you sorry 🙂

**Reply by end-ichi** [reply]
@noccer maybe you have to join the beta tester program, there should be an option when you are on the play store page. Iirc, it might take a little for this to process. If that doesn't work, try tagging iSoron and ask him.
